qcodes/instrument_drivers/stanford_research/SR830_channels.py

```diff
@@ -0,0 +1,602 @@
+            set_cmd=partial(parent._set_ch_display, channel_number),
+            vals=Strings()
+        )
+
+        self.add_parameter(
+            'databuffer',
+            channel=channel_number,
+            parameter_class=ChannelBuffer
+        )
+
+
+class ChannelBuffer(ArrayParameter):
+    """
+    Parameter class for the two channel buffers
+
+    Currently always returns the entire buffer
+    TODO (WilliamHPNielsen): Make it possible to query parts of the buffer.
+    The instrument natively supports this in its TRCL call.
+    """
+
+    def __init__(self, name: str, channel_array: 'SR830Channel', channel: int):
+        """
+        Args:
+            name (str): The name of the parameter
+            channel_array (SR830Channel): The channel array of the parent instrument
+            channel (int): The relevant channel (1 or 2). The name should
+                should match this.
+        """
+        self._valid_channels = (1, 2)
+
+        if channel not in self._valid_channels:
+            raise ValueError('Invalid channel specifier. SR830 only has '
+                             'channels 1 and 2.')
+
+        if not isinstance(channel_array, SR830Channel):
+            raise ValueError('Invalid parent instrument. ChannelBuffer '
+                             'can only live on a channel buffer of SR830.')
+
+        super().__init__(name,
+                         shape=(1,),  # dummy initial shape
+                         unit='V',  # dummy initial unit
+                         setpoint_names=('Time',),
+                         setpoint_labels=('Time',),
+                         setpoint_units=('s',),
+                         docstring='Holds an acquired (part of the) '
+                                   'data buffer of one channel.')
+
+        self.channel = channel
+        self._instrument = channel_array.parent
+
+    def prepare_buffer_readout(self):
+        """
+        Function to generate the setpoints for the channel buffer and
+        get the right units
+        """
+
+        N = self._instrument.buffer_npts()  # problem if this is zero?
+        # TODO (WilliamHPNielsen): what if SR was changed during acquisition?
+        SR = self._instrument.buffer_SR()
+        if SR == 'Trigger':
+            self.setpoint_units = ('',)
+            self.setpoint_names = ('trig_events',)
+            self.setpoint_labels = ('Trigger event number',)
+            self.setpoints = (tuple(np.arange(0, N)),)
+        else:
+            dt = 1/SR
+            self.setpoint_units = ('s',)
+            self.setpoint_names = ('Time',)
+            self.setpoint_labels = ('Time',)
+            self.setpoints = (tuple(np.linspace(0, N*dt, N)),)
+
+        self.shape = (N,)
+
+        params = self._instrument.parameters
+        # YES, it should be: "is not 'none'" NOT "is not None"
+        if params['ch{}_ratio'.format(self.channel)].get() is not 'none':
```

Comment:
Now that the driver uses channels, this line no longer works. It should be something like `self._instrument.submodules['ch{}'.format(self.channel)].ratio.get()`